pkg/agent/client.go

```diff
@@ -79,6 +80,62 @@ func newConnectionManager() *connectionManager {
 	}
 }
 
+// AgentIdentifiers stores identifiers that will be used by the server when
+// choosing agents
+type AgentIdentifiers struct {
+	IPv4        []string
+	IPv6        []string
+	Host        []string
+	CIDR        []string
```

Comment:
@cheftako I understand eventually we want to support CIDR and FailureZone options, but this isn't currently used in the PR. Can we remove it for now to avoid misuse?

Supporting these options should be done on the proxy server/kube-apiserver side. I'm not sure if an agent would even be able to provide this information without querying KAS.

Reply:
Sure. Though I would guess @charleszheng44 may want CIDR sooner rather than later. Pods usually have their IP assigned from the CIDR range for a given node. So if you want to ensure your requests are sent to exactly the right node and you have requests for pods then you will need to do a CIDR match.

Reply:
Oh, so CIDR range is something that can be directly inferred from a pod IP? If that's the case this seems like a pretty big performance improvement since we can almost guarantee that every request routes to the exact node that contains a pod?

Reply:
@Jefftree Yes. CIDR can be quite useful. For example, the proxy server receives a request with dest IP 192.168.0.1, and there exists an agent with CIDR 192.168.0.0/16 as one of its identifiers, then the server will forward the request to the agent. 

Let's add the FailureZone in a seperate PR. I am still wondering how to pass the FailureZone information through a request?  Does pod/node in different FailureZone can have same IP? We can open an issue to discusses this.

Reply:
Remove the FailureZone IdentifierType for now.